src/core/webview/webviewMessageHandler.ts

```diff
@@ -939,45 +932,39 @@ export const webviewMessageHandler = async (provider: ClineProvider, message: We
 					const { apiConfiguration, customSupportPrompts, listApiConfigMeta, enhancementApiConfigId } =
 						await provider.getState()
 
-					// Try to get enhancement config first, fall back to current config
-					let configToUse: ApiConfiguration = apiConfiguration
-					if (enhancementApiConfigId) {
-						const config = listApiConfigMeta?.find((c: ApiConfigMeta) => c.id === enhancementApiConfigId)
-						if (config?.name) {
-							const loadedConfig = await provider.providerSettingsManager.loadConfig(config.name)
```

Comment:
Oops - this looked like a pretty annoying bug. When you enhance the prompt we shouldn't change the active provider profile. We should just fetch it and pass it to `singleCompletionHandler`.